tests/std/tests/GH_004109_iter_value_t_direct_initialization/test.compile.pass.cpp

```diff
@@ -0,0 +1,414 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <algorithm>
+#include <cassert>
+#include <initializer_list>
+#include <iterator>
+#include <memory>
+#include <numeric>
+#include <random>
+
+#ifdef __cpp_lib_execution

```

Comment:
The intent seems to be skipping `/clr:pure`. Should we use `_M_CEE_PURE` or `__cpp_lib_execution`?

Reply:
Well, in my initial commit we could replace `__cpp_lib_execution` with `_M_CEE_PURE ` for more readability. But because we use `usual_matrix.lst` we need to change `__cpp_lib_execution` with testing `_M_CEE_PURE` and ` _HAS_CXX17`. And I think one condition is better than two.